components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -577,12 +583,39 @@ func (g *RTKMovementSensor) Position(ctx context.Context, extra map[string]inter
 	g.ntripMu.Lock()
 	lastError := g.err.Get()
 	if lastError != nil {
-		defer g.ntripMu.Unlock()
-		return geo.NewPoint(0, 0), 0, lastError
+		lastPosition := g.lastposition.GetLastPosition()
+		g.ntripMu.Unlock()
+		if lastPosition != nil {
+			return lastPosition, 0, nil
+		}
+		return nil, 0, lastError
 	}
 	g.ntripMu.Unlock()
 
-	return g.nmeamovementsensor.Position(ctx, extra)
+	position, alt, err := g.nmeamovementsensor.Position(ctx, extra)
+	if err != nil {
+		// Use the last known valid position if current position is (0,0)
+		if position != nil && (position.Lng() == 0 && position.Lat() == 0) {
```

Comment:
Can this not return `math.Nan` from the underlying nmea object and is that what we want to check?